torchchat/utils/quantize.py

```diff
@@ -110,23 +125,73 @@ def quantize_model(
 
     if isinstance(quantize_options, str):
         quantize_options = json.loads(quantize_options)
-
     for quantizer, q_kwargs in quantize_options.items():
```

Comment:
@dillondesilva Small detail regarding "experimental:shared" that @metascroy brought up is that the order of operation matter for this particular quant

We can just do this to enforce it
```suggestion
    # "experimental:shared" Must be executed first
    shared_experimental_quant = quantize_options.pop("experimental:shared", None)
    if shared_experimental_quant is not None:
        quantize_options = {"experimental:shared": shared_experimental_quant} | quantize_options

    for quantizer, q_kwargs in quantize_options.items():
```

Reply:
Or you use the following code to reorder all quantization options in the order they are in quantizer_class_dict
(and adjust that dict to be in the order you want - which right now is just move this experimental:shared quantizer to the top):

```
ordered_quantize_options = { } 

for quantizer in quantizer_class_dict.keys():
  if quantizer in quantize_options:
    ordered_quantize_options |= { quantizer : quantize_options.pop(quantizer) }

if len(quantize_options:) != 0:
  raise RuntimeError(f"unknown quantizer {quantizer} specified")

quantize_options = ordered_quantize_options
```

PS: This is written to subsume the check of line 129-132 because it raises on any unknown quantizers.